src/Orleans.CodeGenerator/Model/SerializableTypeDescription.cs

```diff
@@ -194,7 +194,7 @@ public bool IsPartial
 
         public List<INamedTypeSymbol> SerializationHooks { get; }
 
-        public bool IsImmutable => IsEnumType || Type.HasAnyAttribute(_libraryTypes.ImmutableAttributes);
+        public bool IsShallowCopyable => IsEnumType || !Type.HasBaseType(_libraryTypes.Exception) && _libraryTypes.IsShallowCopyable(Type);
```

Comment:
Is there an explanation for the divergence between this and LibraryTypes.IsShallowCopyable? I wonder if these clauses should be wound into LibraryTypes.IsShallowCopyable (implying that Exception would be marked as not shallow-copyable, which is probably for the best)

Reply:
The previous naming/logic didn't match actual use in code generation, so I changed it to align more with `IsShallowCopyable`, but not for exceptions.

Exceptions seem to be treated generally as shallow-copyable (which makes sense - probably don't want to make local copies of exceptions before serialization for example), but they aren't actually guaranteed to be immutable (but in practice should not be modified carelessly). I left this additional condition here so that a deep-copier is generated for exceptions, so if anyone actually wants a deep copy, it will work. However, the only use-case I found in Orleans code was for legacy net48 exception serialization, which I removed.

Reply:
> they aren't actually guaranteed to be immutable (but in practice should not be modified carelessly)

Yes, unfortunately they aren't... and I see folks passing custom data structures via their exceptions, too